services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -554,26 +551,29 @@ async def stop_computation(
             )
 
         # get run details if any
-        last_run: CompRunsAtDB | None = None
-        with contextlib.suppress(ComputationalRunNotFoundError):
-            last_run = await comp_runs_repo.get(
-                user_id=computation_stop.user_id, project_id=project_id
+        async def _create_computation_response() -> ComputationGet:
+            last_run: CompRunsAtDB | None = None
+            with contextlib.suppress(ComputationalRunNotFoundError):
+                last_run = await comp_runs_repo.get(
+                    user_id=computation_stop.user_id, project_id=project_id
+                )
+
+            return ComputationGet(
+                id=project_id,
+                state=pipeline_state,
+                pipeline_details=await compute_pipeline_details(
+                    complete_dag, pipeline_dag, tasks
+                ),
+                url=TypeAdapter(AnyHttpUrl).validate_python(f"{request.url}"),
+                stop_url=None,
+                iteration=last_run.iteration if last_run else None,
+                result=None,
+                started=compute_pipeline_started_timestamp(pipeline_dag, tasks),
+                stopped=compute_pipeline_stopped_timestamp(pipeline_dag, tasks),
+                submitted=last_run.created if last_run else None,
             )
 
-        return ComputationGet(
-            id=project_id,
-            state=pipeline_state,
-            pipeline_details=await compute_pipeline_details(
-                complete_dag, pipeline_dag, tasks
-            ),
-            url=TypeAdapter(AnyHttpUrl).validate_python(f"{request.url}"),
-            stop_url=None,
-            iteration=last_run.iteration if last_run else None,
-            result=None,
-            started=compute_pipeline_started_timestamp(pipeline_dag, tasks),
-            stopped=compute_pipeline_stopped_timestamp(pipeline_dag, tasks),
-            submitted=compute_pipeline_submitted_timestamp(pipeline_dag, tasks),
-        )
+        return await _create_computation_response()
```

Comment:
Q: out of curiosity why is this wrapped in a function call? could it not have worked without defining a new function?

Reply:
that is a mistake. thanks for checking

Reply:
the last push from the train failed obviously